homeassistant/components/zone/__init__.py

```diff
@@ -37,27 +39,69 @@
 }, extra=vol.ALLOW_EXTRA)
 
 
+def create_zone(hass, name, latitude, longitude, radius, icon=None,
+                passive=DEFAULT_PASSIVE, entity_id=None, update=True):
+    """Create new zone or update an existing one."""
+    return run_coroutine_threadsafe(
+        async_create_zone(
+            hass, name, latitude, longitude, radius, icon, passive, entity_id,
+            update),
+        hass.loop).result()
+
+
+async def async_create_zone(hass, name, latitude, longitude, radius, icon=None,
+                            passive=DEFAULT_PASSIVE, entity_id=None,
+                            update=True):
+    """Create new zone or update an existing one."""
+    zones = hass.data[DOMAIN]
+    if entity_id:
+        object_id = split_entity_id(entity_id)[1]
+        zone = zones.get(object_id)
+        if zone:
+            if update:
+                zone.change(name, latitude, longitude, radius, icon, passive)
+                await zone.async_update_ha_state()
+            return entity_id
+    else:
+        entity_id = async_generate_entity_id(
+            ENTITY_ID_FORMAT, name, None, hass)
+        object_id = split_entity_id(entity_id)[1]
+    zone = Zone(hass, name, latitude, longitude, radius, icon, passive)
+    zone.entity_id = entity_id
+    await zone.async_update_ha_state()
+    zones[object_id] = zone
+    return entity_id
+
+
+def remove_zone(hass, entity_id):
```

Comment:
drop this.

Reply:
The sync version of remove, or the code that removes zones entirely?

Reply:
sync version